api/debug/flags.go

```diff
@@ -136,33 +187,106 @@ func CreateLogDir(logDir string) {
+	var ostream log.Handler
+	output := io.Writer(os.Stderr)
+	if ctx.GlobalBool(logjsonFlag.Name) {
+		ostream = log.StreamHandler(output, log.JsonFormat())
+	} else {
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) && os.Getenv("TERM") != "dumb")
+		if usecolor {
+			output = colorable.NewColorableStderr()
+		}
+		ostream = log.StreamHandler(output, log.TerminalFormat(usecolor))
+	}
+	glogger.SetHandler(ostream)
+
 	// logging
-	log.PrintOrigins(ctx.GlobalBool(debugFlag.Name))
+	debug := ctx.GlobalBool(debugFlag.Name)
+	if ctx.GlobalIsSet(legacyDebugFlag.Name) {
+		debug = ctx.GlobalBool(legacyDebugFlag.Name)
+		logger.Warn("The flag --debug is deprecated and will be removed in the future (v1.12.0), please use --log.debug")
+	}
+	log.PrintOrigins(debug)
 	log.ChangeGlobalLogLevel(glogger, log.Lvl(ctx.GlobalInt(verbosityFlag.Name)))
 	glogger.Vmodule(ctx.GlobalString(vmoduleFlag.Name))
-	glogger.BacktraceAt(ctx.GlobalString(backtraceAtFlag.Name))
+
+	backtrace := backtraceAtFlag.Value
+	if b := ctx.GlobalString(legacyBacktraceAtFlag.Name); b != "" {
+		backtrace = b
+		logger.Warn("The flag --backtrace is deprecated and will be removed in the future (v1.12.0), please use --log.backtrace")
+	}
+	if b := ctx.GlobalString(backtraceAtFlag.Name); b != "" {
+		backtrace = b
+	}
+	glogger.BacktraceAt(backtrace)
 	log.Root().SetHandler(glogger)
 
 	// profiling, tracing
-	runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
-	Handler.SetBlockProfileRate(ctx.GlobalInt(blockprofilerateFlag.Name))
+	runtime.MemProfileRate = memprofilerateFlag.Value
+	if ctx.GlobalIsSet(legacyMemprofilerateFlag.Name) {
+		runtime.MemProfileRate = ctx.GlobalInt(legacyMemprofilerateFlag.Name)
+		logger.Warn("The flag --memprofilerate is deprecated and will be removed in the future (v1.12.0), please use --pprof.memprofilerate")
+	}
+	if ctx.GlobalIsSet(memprofilerateFlag.Name) {
+		runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
+	}
+
+	blockProfileRate := blockprofilerateFlag.Value
+	if ctx.GlobalIsSet(legacyBlockprofilerateFlag.Name) {
+		blockProfileRate = ctx.GlobalInt(legacyBlockprofilerateFlag.Name)
+		logger.Warn("The flag --blockprofilerate is deprecated and will be removed in the future (v1.12.0), please use --pprof.blockprofilerate")
+	}
+	if ctx.GlobalIsSet(blockprofilerateFlag.Name) {
+		blockProfileRate = ctx.GlobalInt(blockprofilerateFlag.Name)
+	}
+	Handler.SetBlockProfileRate(blockProfileRate)
+
 	if traceFile := ctx.GlobalString(traceFlag.Name); traceFile != "" {
 		if err := Handler.StartGoTrace(traceFile); err != nil {
 			return err
 		}
 	}
-	if cpuFile := ctx.GlobalString(cpuprofileFlag.Name); cpuFile != "" {
+	cpuFileName := cpuprofileFlag.Name
+	if ctx.GlobalIsSet(legacyCpuprofileFlag.Name) {
+		cpuFileName = legacyCpuprofileFlag.Name
+		logger.Warn("The flag --cpuprofile is deprecated and will be removed in the future (v1.12.0), please use --pprof.cpuprofile")
+	}
+	if cpuFile := ctx.GlobalString(cpuFileName); cpuFile != "" {
```

Comment:
Unlike above change, it does not check cpuProfileFlag setting after checking legacy flag. Do you have any reason for this?

Reply:
I think It check cpuProfileFlag in the 254th.. 
Could you explain a little more?

Reply:
The code is updated with the function, so this comment is not available anymore. Never mind.